06_PDDE_Checker.R

```diff
@@ -437,23 +469,115 @@ overlapping_hmis_participation <- HMISParticipation %>%
+  ) %>%
+  merge_check_info(checkIDs = 135) %>% 
+  mutate(Detail = 
+           paste0("Bed Type incompatible with Housing Type:  Facility-based beds should align to the Housing Type of site-based and voucher-based beds should align to the Housing Type of tenant-based.")
+  ) %>%
+  select(all_of(PDDEcols))
+
+
+# Project CoC Missing Bed Inventory -----------------------------------
+# Throw a warning if there is no inventory record for a ProjectID and COCCode combo in the ProjectCoC data
+
+ProjectCoC <- ProjectCoC %>% mutate(iy=1)
+
+Active_Inventory_per_COC <- activeInventory %>%
+  mutate(ix=1) %>% 
+  merge(ProjectCoC, by = c("ProjectID", "CoCCode"), all=TRUE) %>%
+  mutate(mer = case_when(ix==1&iy==1~'both',
+                         ix==1~'only_x',
+                         iy==1~'only_y')) %>%
+  filter(mer=="only_y") %>% 
+  merge_check_info(checkIDs = 136) %>% 
+  mutate(Detail = 
+           paste0("Residential projects must have a bed inventory for each CoC they serve.")
+  ) %>%
+  select(all_of(PDDEcols))
+
+ProjectCoC <- ProjectCoC %>% subset(select = -iy)
+  
+  
+# Incorrect CoC in bed inventory -----------------------------------
+# Throw an error if there is no COC record for a ProjectID and COCCode combo in the inventory data
+ProjectCoC <- ProjectCoC %>% mutate(iy=1)
+
+COC_Records_per_Inventory <- activeInventory %>%
+  mutate(ix=1) %>% 
+  merge(ProjectCoC, by = c("ProjectID", "CoCCode"), all=TRUE) %>%
+  mutate(mer = case_when(ix==1&iy==1~'both',
+                         ix==1~'only_x',
+                         iy==1~'only_y')) %>% 
+  filter(mer=="only_x") %>% 
+merge_check_info(checkIDs = 137) %>%
+  mutate(Detail = 
+           paste0("Any CoC represented in a project's active bed inventory records must also be listed as a CoC associated with the Project.")
+  ) %>%
+  select(all_of(PDDEcols))
+
+ProjectCoC <- ProjectCoC %>% subset(select = -iy)
+
+
+# More units than beds in inventory record. -----------------------------------
+more_units_than_beds_inventory <- activeInventory %>%
+  filter(UnitInventory > BedInventory) %>% 
+merge_check_info(checkIDs = 138) %>%
+  mutate(Detail = 
+           paste0("An inventory record cannot have more units than total number of beds. Please update this inventory record in HMIS to ensure that units are less than or equal to the number of beds.")
+  ) %>%
+  select(all_of(PDDEcols))
+
+
+# Client-level data in VSP organization. -----------------------------------
+# Projects under Organizations marked as Victim Service Providers should not have client data in HMIS
+# If VictimServiceProvider==1, then flag as high priority error if client data is present (i.e. any enrollments).
+
+vsp_clients <- Organization %>%
+  select(OrganizationID, VictimServiceProvider) %>% 
+  filter(VictimServiceProvider==1) %>%
+  left_join(Project0(), by = "OrganizationID") %>%
+  left_join(Enrollment, by = "ProjectID") %>%
```

Comment:
I don't think you need `Organization` here. `Project0` contains all the identifying PDDECols, including `VictimServiceProvider`. See line 78 of the 04_initial_data_prep.R script:

```
Project0(project_prep %>%
  select(ProjectID,
         ProjectName,
         OrganizationID,
         OrganizationName,
         OperatingStartDate,
         OperatingEndDate,
         ProjectType,
         RRHSubType,
         VictimServiceProvider) %>%
  unique()
)
```

Also, I think the problem with left-joining `Enrollment` is that the PDDE export Detail will go from being Project-level to Enrollment-level. So you'll either have to dedup at the end or figure out a different way to check for client data.

Reply:
Done